lib/boot-files.js

```diff
@@ -103,13 +106,27 @@ function writeBootFiles(opts, callback) {
 
         // Write out networking.json file.
         function _bootTimeFile(arg, cb) {
-            mod_net_file.write(arg, function (err) {
+            mod_net_file.write(arg, function (_err) {
                 // Don't let this hold up booting - we can always fall back to
                 // parameters
                 return cb();
             });
         },
 
+        // Write out any suplemental boot module files we need for this CN
+        function _bootTimeModulesFs(arg, cb) {
+            // Skip if no suplemental boot file modules
+            if (!arg.bootParams.boot_modules) {
+                cb();
+                return;
+            }
+            bootModuleFiles.write(arg, function (_err) {
+                // Don't let this hold up booting - we can always fall back to
+                // parameters
```

Comment:
Is this true?  While the content of `networking.json` may be duplicated in boot args, I don't think any other module is.  For Linux, I don't expect that any boot module will be redundant.

Reply:
I bet we'll have to make some differences for linux. Anyway, let's error out if there's any error here.